src/components/todoItem.js

```diff
@@ -1,22 +1,31 @@
 import React from 'react';
 import {observer} from 'mobx-react';
+import {observable, expr} from 'mobx';
 
 const ESCAPE_KEY = 27;
 const ENTER_KEY = 13;
 
+
+class State {
+	@observable editText = '';
+	constructor(editText){
+		this.editText = editText;
+	}
+}
+
 @observer
 export default class TodoItem extends React.Component {
 	constructor(props, context) {
 		super(props, context);
-		this.state = {editText: props.todo.title};
+		this.state = new State(props.todo.title);
```

Comment:
I would not put this into `this.state`, as that would start interference / break react's own setState mechanism which would otherwise work. Anything other field would do I think. You could even introduce `@observe editText = ''` directly as field on the component